lld/ELF/Writer.cpp

```diff
@@ -1444,6 +1444,38 @@ static void finalizeSynthetic(Ctx &ctx, SyntheticSection *sec) {
   }
 }
 
+static bool canInsertPadding(OutputSection *sec) {
+  StringRef s = sec->name;
+  return s == ".bss" || s == ".data" || s == ".data.rel.ro" || s == ".rodata" ||
+         s.starts_with(".text");
+}
+
+static void shufflePadding(Ctx &ctx) {
+  std::mt19937 g(*ctx.arg.shufflePadding);
+  PhdrEntry *curPtLoad = nullptr;
+  for (OutputSection *os : ctx.outputSections) {
+    if (!canInsertPadding(os))
+      continue;
+    for (SectionCommand *bc : os->commands) {
+      if (auto *isd = dyn_cast<InputSectionDescription>(bc)) {
+        SmallVector<InputSection *, 0> tmp;
+        if (os->ptLoad != curPtLoad) {
```

Comment:
I think it will be worth checking to see if there is an explicit address expression associated with the output section before doing this. The implication being that someone could want the OutputSection at a specific address and adding padding could cause problems. Not likely a problem for an Linux style executable or shared object, but possible for linker scripts.

It may also be worth a quick check to see if there are any symbol assignments within the OutputSection.
```
MYOS : { MYOS_start = . ; *(.text.myos) ; MYOS_end = .; } 
```
I think you are likely to be aiming at the default no linker script case where this isn't likely to happen. Could be worth just disabling ShufflePadding if anything other than InputSectionDescriptions are found?

Reply:
In the case you mention it won't necessarily be the case that start/stop symbols will break things. The distance between `MYOS_start` and `MYOS_end` will increase but a normal program shouldn't be able to notice this (modulo things like flash size limitations) just like it shouldn't be able to notice that the compiler was replaced with one that generates slightly larger code. Silently disabling the feature could lead to confusion so I'd prefer not to do that given that the user is opting into the feature with a flag.